### DISCUSSION TITLE
How to use  rowcount, rownum, colcount, and colnum variables in hooks

### DISCUSSION THREAD

**Discussion by kalekje** [lvjr/tabularray#557] [Q&A]
I would like to use these variables to create a custom TableCommand. How do I access these variables? Please see MWE below.

```

\documentclass{scrartcl}
\usepackage{luacode}
\usepackage{tabularray}

\UseTblrLibrary{hook}



\AddToHook{tabularray/cell/before}{-}
\AddToHook{tabularray/cell/after}{-}

\begin{document}

\begin{tblr}{|l|l|l|}
  rownum, colnum & rownum, colnum & rownum, colnum\\
\end{tblr}

\end{document}

```

**Comment by lvjr** [comment]
For questions, it is better to use GitHub Discussions instead of GitHub Issues.

It is unclear what you want to get from your example. But these counters are normal LaTeX counters, so you can write code like `\arabic{rownum}` and `\value{colcount}`.

**Reply by kalekje** [reply]
Thank you for the reply. That helps. Would you happen to know how I can access this in Lua? Typically, I would use a `tex.count['rownum']`, but I am getting a counter undefined problem.

```
\documentclass{scrartcl}
\usepackage{luacode}
\usepackage{tabularray}

\UseTblrLibrary{hook}


%\AddToHook{tabularray/cell/before}{-}
%\AddToHook{tabularray/cell/after}{-}

\begin{document}

\begin{tblr}{|l|l|l|}
  \arabic{rownum} &  &  \\
  \luadirect{texio.write_nl(1)} &  &  \\
  \luadirect{texio.write_nl(tex.count['rowcount'])} &  &  \\
\end{tblr}

\end{document}
```

**Reply by kalekje** [reply]
Nvm, I just needed to prepend the counter with c@.
